contract/src/main/java/io/neow3j/contract/NonFungibleToken.java

```diff
@@ -92,17 +139,66 @@ public TransactionBuilder transfer(Wallet wallet, Hash160 to, byte[] tokenID,
 
     /**
      * Gets the owner of the token with {@code tokenId}.
+     * <p>
+     * This method is intended to be used for non-divisible NFTs only.
      *
      * @param tokenId the token ID.
-     * @return a list of owners of the token.
+     * @return the token owner.
      * @throws IOException if there was a problem fetching information from the Neo node.
      */
     public Hash160 ownerOf(byte[] tokenId) throws IOException {
+        if (getDecimals() != 0) {
+            throw new IllegalStateException("This method is only intended for non-divisible NFTs.");
+        }
         return callFunctionReturningScriptHash(OWNER_OF, byteArray(tokenId));
     }
 
+    // Divisible NFT methods
+
+    public TransactionBuilder transferDivisible(Wallet wallet, Hash160 from, Hash160 to,
+            BigInteger amount, byte[] tokenID) throws IOException {
+        return transferDivisible(wallet, from, to, amount, tokenID, null);
+    }
+
+    public TransactionBuilder transferDivisible(Wallet wallet, Hash160 from, Hash160 to,
+            BigInteger amount, byte[] tokenID, ContractParameter data) throws IOException {
+        if (!wallet.holdsAccount(from)) {
+            throw new IllegalArgumentException("The provided wallet does not contain the from " +
+                    "account.");
+        }
+        if (getDecimals() == 0) {
+            throw new IllegalStateException("This method is only intended for divisible NFTs.");
+        }
+
+        return invokeFunction(TRANSFER, hash160(from), hash160(to), integer(amount),
+                byteArray(tokenID), data)
+                .wallet(wallet)
+                .signers(calledByEntry(from));
+    }
+
```

Comment:
Please add Javadoc to these methods and consider renaming them to simply `transfer`. I think it's enough if the javadoc says that they are intended for divisible tokens only.